### ISSUE TITLE
Onlykey not detected with Snap and Appimage

### ISSUE THREAD

**Issue by bpycinski** [keepassxreboot/keepassxc#5226]
## Overview
The program cannot detect an OnlyKey (this concerns only Snap and AppImage packages on Linux)

## Context
I am running Linux Mint 20 (kernel 5.4.0-42-generic). Both Snap (2.6.0 2020-07-07) and AppImage (2.6.0) versions cannot connect to OnlyKey for unlocking or creating a database ("No hardware keys detected"). There are no issues with an YubiKey 5 NFC (i.e. the YK is detected and works fine). The program compiled by myself from sources (2.6.0, gcc 9.3.0) works fine with OnlyKey and YubiKey. I see that libgcrypt versions differ, but I do not know how to upgrade them by myself (see listings below).

By the way, the same keys tested on Windows machine work fine with KeepassXC.

## Snap version - not working
KeePassXC - Version 2.6.0
Revision: 0765954
Distribution: Snap

Qt 5.9.5
Debugging mode is disabled.

Operating system: Ubuntu Core 18
CPU architecture: x86_64
Kernel: linux 5.4.0-42-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.1

## AppImage version - not working
KeePassXC - Version 2.6.0
Revision: 0765954
Distribution: AppImage

Qt 5.14.2
Debugging mode is disabled.

Operating system: Linux Mint 20
CPU architecture: x86_64
Kernel: linux 5.4.0-42-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.1

## Compiled version - working
KeePassXC - Version 2.6.0
Revision: 0765954

Qt 5.12.8
Debugging mode is disabled.

Operating system: Linux Mint 20
CPU architecture: x86_64
Kernel: linux 5.4.0-42-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- YubiKey

Cryptographic libraries:
 libgcrypt 1.8.5

**Comment by droidmonkey** [comment]
I'm surprised the appimage is failing, but the snap fails because you did not give it permission. See https://keepassxc.org/docs/#faq-appsnap-homedir

You want the access USB directly permission.

**Comment by bpycinski** [comment]
Hi, thanks for your reply. The permission has been set before:
```
$ snap connections keepassxc
Interface              Plug                       Slot                            Notes
content[gtk-3-themes]  keepassxc:gtk-3-themes     gtk-common-themes:gtk-3-themes  -
content[icon-themes]   keepassxc:icon-themes      gtk-common-themes:icon-themes   -
content[sound-themes]  keepassxc:sound-themes     gtk-common-themes:sound-themes  -
desktop                keepassxc:desktop          :desktop                        -
desktop-legacy         keepassxc:desktop-legacy   :desktop-legacy                 -
gsettings              keepassxc:gsettings        :gsettings                      -
home                   keepassxc:home             :home                           -
network                keepassxc:network          :network                        -
network-bind           keepassxc:network-bind     :network-bind                   -
opengl                 keepassxc:opengl           :opengl                         -
raw-usb                keepassxc:raw-usb          :raw-usb                        manual
removable-media        keepassxc:removable-media  -                               -
unity7                 keepassxc:unity7           :unity7                         -
wayland                keepassxc:wayland          :wayland                        -
x11                    keepassxc:x11              :x11                            -
```

Moreover, all the versions can properly read challenge-responce from a YubiKey 5.

**Comment by droidmonkey** [comment]
Do you have the yubikey udev package installed? Easy test is run the snap with sudo and see if it sees your key. If so, you need the udev package installed.

**Comment by bpycinski** [comment]
Yes, the package is installed. If I wasn't clearly, Yubikey works with all types of packages. 
Onlykey is broken only with snap and appimage. In the self-compiled version, it works. 
Running the application with sudo doesn' t change anything.

**Comment by droidmonkey** [comment]
That makes no sense to me.

**Comment by droidmonkey** [comment]
Arggghhh, this must be because the Snap and AppImage packages are using ykpers 1.19 or less.

**Comment by bpycinski** [comment]
Hi, indeed, the compiled version is linked to `1.20.0`, the snap version to `1.18.0` and the appimage version (probably, as there are no symlinks in the package) to `1.17.2`.

**Comment by droidmonkey** [comment]
Fixed in 2.7.0
